src/main/java/com/maddyhome/idea/vim/group/NotificationService.kt

```diff
@@ -95,7 +97,8 @@ internal class NotificationService(private val project: Project?) {
         "set ideajoin",
         "idejoin"
       ) {
-        injector.globalIjOptions().ideajoin = true
+        // 'ideajoin' is global-local, so we'll set it as a global value
+        injector.optionGroup.setOptionValue(IjOptions.ideajoin, OptionScope.GLOBAL, VimInt.ONE)
```

Comment:
Mainly because this is a global-local option which we don't currently support properly (but will in the next PR, currently in progress 😁). 

The accessors are aimed at the common case of getting/setting the "effective" value of an option, but we don't really support "effective" scope in the API, only `GLOBAL` and `LOCAL`, and this makes it hard to handle global-local options correctly. Global-local options are usually global, but can be overridden with a local value, so the `ideajoin` accessor is a property on the `EffectiveIjOptions` class, with a scope of `LOCAL(editor)`. Getting the value from the accessor works fine, as a side effect of the implementation - because we lazily initialise options, when the local value isn't set, we fall back to the global value. But set will set the value at `LOCAL` scope, which is not what we want here.

So for now, we need to explicitly set it at `GLOBAL` scope via `setOptionValue`.

I'm introducing an `AUTO` scope in the next PR which is the "effective" scope of an option, and the option accessors will use this to get/set at the right scope. The scopes now map closely to `:set`, `:setglobal` and `:setlocal`, and the API behaves the same as the commands. Given `AUTO` scope and a global-local option, `getOptionValue` (and the accessors) will return the local value if set, global otherwise. `setOptionValue` follows Vim behaviour and updates the global value, and resets/updates the local value. (It's important to update the global value so that we can eagerly initialise options for new windows, which is itself important for options that are required at window initialisation time, such as, finally, `'wrap'` 😅)

All of which means that with the next PR, we can use the accessors API to set this value.